lib/system/sysio.nim

```diff
@@ -223,34 +222,42 @@ proc endOfFile(f: File): bool =
   return c < 0'i32
   #result = c_feof(f) != 0
 
-proc readAllFile(file: File, len: int): string =
+proc readAllFile(file: File, len: int, result: var string) =
   # We acquire the filesize beforehand and hope it doesn't change.
   # Speeds things up.
-  result = newString(len)
   let bytes = readBuffer(file, addr(result[0]), len)
   if endOfFile(file):
     if bytes < len:
       result.setLen(bytes)
   else:
     # We read all the bytes but did not reach the EOF
     # Try to read it as a buffer
-    result.add(readAllBuffer(file))
+    readAllBuffer(file, result)
 
-proc readAllFile(file: File): string =
-  var len = rawFileSize(file)
-  result = readAllFile(file, len)
-
-proc readAll(file: File): TaintedString =
+proc readAll*(file: File, result: var TaintedString) =
+  ## Reads all data from the stream `file` into `result` which may be nil.
```

Comment:
There is at least one proc in the stdlib that assumes a ``var string`` parameter isn't nil. From an efficiency point of view it's more efficient to assume this. 

In either case, it would be good to be consistent here.